sdk/src/main/scala/com/horizen/account/state/AccountStateView.scala

```diff
@@ -151,25 +151,43 @@ class AccountStateView(
     // Tx context for stateDB, to know where to keep EvmLogs
     setupTxContext(txHash, txIndex)
 
-    // apply message to state
-    val status = try {
-      applyMessage(msg, blockGasPool)
-      ReceiptStatus.SUCCESSFUL
-    } catch {
-      // any other exception will bubble up and invalidate the block
-      case err: ExecutionFailedException =>
-        log.error(s"applying message failed, tx.id=${ethTx.id}", err)
-        ReceiptStatus.FAILED
-    } finally {
-      // finalize pending changes, clear the journal and reset refund counter
+    val revisionId = snapshot
+    val initialBlockGas = blockGasPool.getGas
+    try {
+      // apply message to state
+      val status = try {
+        applyMessage(msg, blockGasPool)
+        ReceiptStatus.SUCCESSFUL
+      } catch {
+        // any other exception will bubble up and invalidate the block
+        case err: ExecutionFailedException =>
+          log.error(s"applying message failed, tx.id=${ethTx.id}", err)
+          ReceiptStatus.FAILED
+      } finally {
+        // finalize pending changes, clear the journal and reset refund counter
+        // make sure we disable automatic gas consumption in case a message processor enabled it
+        disableGasTracking()
+      }
+      val consensusDataReceipt = new EthereumConsensusDataReceipt(
+        ethTx.version(), status.id, blockGasPool.getUsedGas, getLogs(txHash))
+      log.debug(s"Returning consensus data receipt: ${consensusDataReceipt.toString()}")
+      consensusDataReceipt
```

Comment:
what about the `null` logs case here? I think we need to solve this issue first of all.

Reply:
Do you mean the null value returned from the StateDb? If it is, I have fixed it in the orphan txs branch.